### DISCUSSION TITLE
Build IPA file from Windows PC / IntelliJ

### DISCUSSION THREAD

**Discussion by deesnay** [airsdk/Adobe-Runtime-Support#2883] [General]
Hi,
It looks like I should still be able to compile an IPA on PC, right? But I can't.
I have an AIR project in IntelliJ, based on AIR SDK 33.1.1.935 (free demo version from Harman).
It's an existing AIR project that is usually built on Mac OS, with this same SDK (paid version with a license), successfully.
I'm trying to get it work on Windows.
I've tried to compile in debug mode as well as in release mode. Both fail to install.

In debug mode: I launch iTunes, wait until the iPad is detected, and then launch debugging from IntelliJ. Compilation starts, the iPad is detected, but the process fails at some point with the following warnings & errors:

WARNING: could not locate iTunesMetadata.plist in archive!  
WARNING: could not locate Payload/PharmaVisite.app/SC_Info/PharmaVisite.sinf in archive! 
ERROR: Install failed. Got error "ApplicationVerificationFailed" with code 0xe8008017: Failed to verify code signature of /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.4bLB7o/extracted/Payload/PharmaVisite.app : 0xe8008017 (A signed resource has been added, modified, or deleted.) 

I suppose the warnings don't block the process? But the error does?
I've found similar issues in other posts, except for the last part; "(A signed resource has been added, modified, or deleted.)"

In release mode:

I can compile the ipa successfully, and then drag it to the iPad via iTunes. The installation process seems to work fine, except the actual app icon is missing (generic icon istead). When I try to open the app, it fails, with a warning stating: "Unable to install the app. It cannot be installed because its integrity could not be verified."

The app uses an inHouse certificate.
And also ANEs.

I'm not sure where the problem lies: certificates? ANEs? IntelliJ configuration?...
Any help is welcome. Thanks in advance.

**Comment by ajwfrost** [comment]
Hi

Sorry for the delay here. You're right that the warnings can be ignored, we always see them. The error is the issue:
```
Failed to verify code signature : 0xe8008017 (A signed resource has been added, modified, or deleted.)
```

This means that the code signature has included a file that's not actually in the package, or vice versa. Knowing Apple and Windows, I suspect the file might be one of these ".DS_Store" ones that crops up from time to time.. are you able to check your Windows folder view settings so that you see all hidden files, and then search your AIR application's folders for any of these files? they come from zip/archive files I believe, you can safely delete them from Windows.

thanks

**Reply by deesnay** [reply]
Thank you very much, it definitely works! :)
Actually, I had tried to filter these files via IntelliJ at compiling, but I think this filtering option doesn't work as expected from me.

What doesn't seem to work now is debugging over USB. The app gets installed alright, but then I have to touch the app icon on the iPad desktop to open it, and it doesn't seem to connect to the output panel in IntelliJ. But a Mac developer tells me that's the case also when developing on Mac OS...?
